app/app.go

```diff
@@ -259,3 +271,92 @@ func (a *App) HandleBlock(blk *blockchain.Block) {
 		log.Debug("added blk number %d to chain", blk.BlockNumber)
 	}
 }
+
+// SyncBlockchain updates the local copy of the blockchain by requesting missing
+// blocks from peers. Returns error if we are not connected to any peers.
+func (a *App) SyncBlockchain() error {
+	prevHead := a.Chain.Blocks[len(a.Chain.Blocks)-1]
+	newBlockChan := make(chan *blockchain.Block)
+	errChan := make(chan *msg.ProtocolError)
+
+	// Define a handler for responses to our block requests
+	blockResponseHandler := func(blockResponse *msg.Response) {
+		if blockResponse.Error != nil {
+			errChan <- blockResponse.Error
+			return
+		}
+
+		block, ok := blockResponse.Resource.(blockchain.Block)
+		if !ok {
+			newBlockChan <- nil
+			return
+		}
+
+		newBlockChan <- &block
+	}
+
+	// Continually request the block after the latest block in our chain until
+	// we are totally up to date
+	for {
+		lastBlock := a.Chain.Blocks[len(a.Chain.Blocks)-1]
+
+		reqParams := map[string]interface{}{
+			"lastBlockHash": blockchain.HashSum(lastBlock),
+		}
+
+		blockRequest := msg.Request{
+			ID:           uuid.New().String(),
+			ResourceType: msg.ResourceBlock,
+			Params:       reqParams,
+		}
+
+		// Get a list of peer addresses. We will use these to retreive peers from
+		// the PeerStore that we will then send block requests to
+		peerAddrs := a.PeerStore.Addrs()
+		if len(peerAddrs) == 0 {
+			return errors.New("SyncBlockchain failed: no peers to request blocks from")
+		}
+
+		// Pick a peer to send the request to (this won't always be the same one)
+		p := a.PeerStore.Get(peerAddrs[0])
```

Comment:
Yea I agree, I think there should just be a `GetRandom` function for this. But we do need to select a peer to send the request to, because broadcasting a request doesn't really make much sense.